script.js

```diff
@@ -0,0 +1,142 @@
+pickPaper.addEventListener('click', () => { playerPick('paper') });
+pickScissors.addEventListener('click', () => { playerPick('scissors')});
+
+let player = {
+    name: '',
+    score: 0
+};
+
+let computer = {
+    score: 0
+};
+
+function newGame() {
+    const playerNameElem = document.getElementById('js-playerName');
+    if (player.name === '') {
+        player.name = prompt('Please enter your name', 'imię gracza');
+    }
+    hideWinner();
+    if (player.name) {
+        player.score = 0;
+        computer.score = 0;
+        setGameElements('started');
+        playerNameElem.innerHTML = player.name;
+        setGamePoints();
+    }
+}
+
+function setGameElements(gameState, winner) {
+    const newGameElem = document.getElementById('js-newGameElement');
+    const pickElem = document.getElementById('js-playerPickElement');
+    const resultsElem = document.getElementById('js-resultsTableElement');
+    switch (gameState) {
+        case 'started':
+            newGameElem.style.display = 'none';
+            pickElem.style.display = 'block';
+            resultsElem.style.display = 'block';
+            break;
+        case 'ended':
+            newGameBtn.innerText = 'Jeszcze raz';
+            hideLastRoundWinner();
+        default:
+            newGameElem.style.display = 'block';
+            pickElem.style.display = 'none';
+            resultsElem.style.display = 'none';
+    }
+}
+
+function getComputerPick() {
+    const possiblePicks = ['rock', 'paper', 'scissors'];
+    return possiblePicks[Math.floor(Math.random() * 3)];
+}
+
+function playerPick(playerPick) {
+    const playerPickElem = document.getElementById('js-playerPick');
+    const computerPickElem = document.getElementById('js-computerPick');
+    const computerPick = getComputerPick();
+    playerPickElem.innerHTML = playerPick;
+    computerPickElem.innerHTML = computerPick;
+    checkRoundWinner(playerPick, computerPick);
+}
+
+function checkRoundWinner(playerPick, computerPick) {
+    const playerResultElem = document.getElementById('js-playerResult');
+    const computerResultElem = document.getElementById('js-computerResult');
+    const drawElem = document.getElementById('js-draw');
+    let winnerIs = 'player';
+    playerResultElem.innerHTML = '';
+    computerResultElem.innerHTML = '';
+    drawElem.innerHTML = '';
+    
+    if (playerPick == computerPick) {
+        winnerIs = 'noone'; 
+    } else if (
+        (computerPick === 'rock' && playerPick === 'scissors') ||
+        (computerPick === 'scissors' && playerPick === 'paper') ||
+        (computerPick === 'paper' && playerPick === 'rock')) {
+
+        winnerIs = 'computer';
+    }
+
+    if (winnerIs == 'player') {
+        playerResultElem.innerHTML = "Win!";
+        player.score++;
+    } else if (winnerIs === 'computer') {
+        computerResultElem.innerHTML = "Win!";
+        computer.score++;
+    } else {
+        drawElem.innerHTML = "Draw!";
+    }
```

Comment:
logika tej funkcji do uproszczenia. Nie stosuj pomocniczej zmiennej `winnerIs`.

W tym momencie wpisujemy do niej zawartość zależnie od logiki kto wygrał, a potem ją jeszcze raz sprawdzamy żeby podjąć jakieś akcje. Da się to zrobić w jednym kroku. Przez to może powstać Ci trochę powtarzającego się kodu, ale wtedy wydzielisz powtarzające się elementy do pomocniczych funkcji.